CONTRIBUTING.md

```diff
@@ -1,189 +1,272 @@
 
-The e2e test suite can be run from the e2e folder(`cd e2e`) with this command - `go test -v -ginkgo.v`.
-Alternatively, if you [install ginkgo binary](https://github.com/onsi/ginkgo#global-installation), you can run it with `ginkgo -v`.
+**Note:** vCluster won't start until you connect with the debugger.
+**Note:** vCluster will be stopped once you detach your debugger session.
 
-For running conformance tests, please take a look at [conformance tests](https://github.com/loft-sh/tree/vcluster/main/conformance/v1.21)
+### Access your vCluster and Set your local KubeConfig
 
-## Build vcluster CLI tool
+Download the [vCluster CLI](https://www.vcluster.com/docs/get-started/) and use it to connect to your virtual cluster.
 
-Build:
+By connecting to the vCluster using the CLI, you set your local KubeConfig to the virtual cluster
 
 ```
-go generate ./... && go build -o vcluster cmd/vclusterctl/main.go # build vcluster cli
+$ vcluster connect vcluster
 ```
 
-Run:
+## Build and Test the vcluster CLI tool
+
+Build the CLI tool 
 
 ```
-./vcluster create v1 # create vcluster
+$ go generate ./... && go build -o vcluster cmd/vclusterctl/main.go # build vcluster cli
 ```
 
-### Deploy vCluster from source
+Test the built CLI tool
 
-#### 1. Check if you have the following software installed:
+```
+./vcluster create v1 # create vcluster
+``` 
 
-* A running Kubernetes cluster v1.26 or newer and Kubectl installed, verify Kubernetes version via `kubectl version`
-* Docker needs to be installed (e.g. docker-desktop, orbstack, rancher desktop etc.)
-* Helm v3.10.0+ installed and verify via `helm version`
-* Golang version 1.22 and verify via `go version`
+## Developing from Source Code
+
+### Pre-requisites
+
+* [Golang v1.22](https://go.dev/doc/install) 
 
-#### 2. Git clone the repository `git clone https://github.com/loft-sh/vcluster.git`
+### Uninstall vCluster CLI
+
+If you already have vCluster CLI installed, please make sure to uninstall it first.
+
+### Build vCluster CLI from Source Code
 
-#### 3. Build vCluster CLI first:
 ```
-sudo go build -mod vendor -o /usr/local/bin/vcluster cmd/vclusterctl/main.go
+$ sudo go build -mod vendor -o /usr/local/bin/vcluster cmd/vclusterctl/main.go
 ```
 
-**If you already have vCluster CLI installed, please make sure to uninstall it first.**
-
-#### 4. Verify vCluster CLI was compiled correctly via `vcluster version`:
+### Verify vCluster CLI was compiled correctly
 
 ```
 $ vcluster version
 vcluster version 0.0.1
 ```
 
-#### 5. Build vCluster Container Image:
+### Build vCluster Container Image
 
 ```
 docker build . -t my-vcluster:0.0.1
 ```
 
-#### 5a. Optional: if using kind, you need to import the image into kind
+#### Importing vCluster Container Image for kind Users
+
+If using kind as your local Kubernetes cluster, you need to import the image into kind.
 
 ```
-kind load docker-image my-vcluster:0.0.1
+$ kind load docker-image my-vcluster:0.0.1
 ```
 
```

Comment:
what I usually do is push to my own registry and use that image instead, since I don't use kind